### DISCUSSION TITLE
DS213 mod to get up to 8 gpio pins

### DISCUSSION THREAD

**Discussion by Alexxdal** [gabonator/LA104#73] [Ideas]
the DS213 according to the schematic, have 8 unconnected gpio pins that can be used by soldering wires on the stm32 chip, it needs a little bit of soldering skills but with this mod you can have most of the LA104 functions

PE6 -> pin 5
PE5 -> pin 4
PE4 -> pin 3
PE3 -> pin 2
PA15 -> pin 77
PC14 -> pin 8
PA10 -> pin 69
PC4 -> pin 33

![20210731_144345](https://user-images.githubusercontent.com/16624072/127740674-a0ee5ecd-de9e-4e9f-8029-6ccf7fa6c6f8.jpg)

![ds213mod](https://user-images.githubusercontent.com/16624072/127748232-ccc9d84f-8c59-41a7-8b37-be33f2b94fa4.jpg)

**Comment by Alexxdal** [comment]
**I2C Scanner with DS213 mod**

![20210801_164731](https://user-images.githubusercontent.com/16624072/127775267-b7467aac-25ca-46d7-bb66-ad32edc37d29.jpg)

**Comment by Alexxdal** [comment]
**DS1820 Temperature Grapher**

![20210801_165858](https://user-images.githubusercontent.com/16624072/127775611-1cef46e1-3ac9-4195-a4bc-433aae521776.jpg)

**Reply by Pepito-Payet** [reply]
Hello,

Can you also use it as an oscilloscope?

Sincerely,

**Reply by Pepito-Payet** [reply]
> **DS1820 Temperature Grapher**
> 
> ![20210801_165858](https://user-images.githubusercontent.com/16624072/127775611-1cef46e1-3ac9-4195-a4bc-433aae521776.jpg)

Hello, @Alexxdal

Can you also use it as an oscilloscope?

Sincerely,

**Reply by Alexxdal** [reply]
Yes it  work as  normal but you have to be carefull to not draw too much power from the 3volt because you can damage the 3volt regulator like i did

**Reply by Pepito-Payet** [reply]
> Yes it work as normal but you have to be carefull to not draw too much power from the 3volt because you can damange the 3volt regulator like i did

@Alexxdal 

How did you get the firmware to work please ? I am on Windows, I copied the files located in the folder LA104-master\LA104-master\tools\fpga in the DFU mode of the DS213 but it does not work...

**Reply by Alexxdal** [reply]
@Pepito-Payet download this release https://github.com/gabonator/LA104/releases/tag/2020.12.23 then in dfu mode copy the system_ds213.hex file in it and restart the device, next time you connect the device(not in dfu) to usb you should see a folder now simply copy the file from ds213apps.zip

**Reply by Pepito-Payet** [reply]
> @Pepito-Payet download this release https://github.com/gabonator/LA104/releases/tag/2020.12.23 then in dfu mode copy the system_ds213.hex file in it and restart the device, next time you connect the device(not in dfu) to usb you should see a folder now simply copy the file from ds213apps.zip

@Alexxdal 

It works, thank you very much!
But the miniosc.elf does not work...

**Reply by Alexxdal** [reply]
@Pepito-Payet i know i had a version that i fixed but i don't know where it is because i stop working o this project because my ds broke

**Reply by Pepito-Payet** [reply]
> 

@Alexxdal 

Oh dear!

For the WebUsb it also works on windows? I don't see anything, you have to type an ip address in a web page?
